skrub/_dataframe/_common.py

```diff
@@ -0,0 +1,780 @@
+@col.specialize("polars")
+def _col_polars(df, col_name):
+    return df[col_name]
+
+
+@dispatch
+def collect(df):
+    return df
+
+
+@collect.specialize("polars", argument_type="LazyFrame")
+def _collect_polars_lazyframe(df):
+    return df.collect()
+
+
+#
+# Querying and modifying metadata
+# ===============================
+#
+
+
+@dispatch
+def shape(obj):
+    raise NotImplementedError()
+
+
+@shape.specialize("pandas")
+def _shape_pandas(obj):
+    return obj.shape
+
+
+@shape.specialize("polars")
+def _shape_polars(obj):
+    return obj.shape
+
+
+@dispatch
+def name(col):
+    raise NotImplementedError()
+
+
+@name.specialize("pandas")
+def _name_pandas(col):
+    return col.name
+
+
+@name.specialize("polars")
+def _name_polars(col):
+    return col.name
+
+
+@dispatch
+def column_names(df):
+    raise NotImplementedError()
+
+
+@column_names.specialize("pandas")
+def _column_names_pandas(df):
+    return list(df.columns.values)
+
+
+@column_names.specialize("polars")
+def _column_names_polars(df):
+    return df.columns
+
+
+@dispatch
+def rename(col, new_name):
+    raise NotImplementedError()
+
+
+@rename.specialize("pandas")
+def _rename_pandas(col, new_name):
+    return col.rename(new_name)
+
+
+@rename.specialize("polars")
+def _rename_polars(col, new_name):
+    return col.rename(new_name)
+
+
+@dispatch
+def set_column_names(df, new_column_names):
+    raise NotImplementedError()
+
+
+@set_column_names.specialize("pandas")
+def _set_column_names_pandas(df, new_column_names):
+    return df.set_axis(new_column_names, axis=1)
+
+
+@set_column_names.specialize("polars")
+def _set_column_names_polars(df, new_column_names):
+    return df.rename(dict(zip(df.columns, new_column_names)))
+
+
+#
+# Inspecting dtypes and casting
+# =============================
+#
+
+
+@dispatch
+def dtype(column):
+    return column.dtype
+
+
+@dtype.specialize("pandas")
+def _dtype_pandas(column):
+    return column.dtype
+
+
+@dtype.specialize("polars")
+def _dtype_polars(column):
+    return column.dtype
+
+
+@dispatch
+def cast(column, dtype):
+    raise NotImplementedError()
+
+
+@cast.specialize("pandas")
+def _cast_pandas(column, dtype):
+    return column.astype(dtype)
+
+
+@cast.specialize("polars")
+def _cast_polars(column, dtype):
+    return column.cast(dtype)
+
+
+@dispatch
+def pandas_convert_dtypes(obj):
+    return obj
+
+
+@pandas_convert_dtypes.specialize("pandas")
+def _pandas_convert_dtypes_pandas(obj):
+    return obj.convert_dtypes()
+
+
+@dispatch
+def is_bool(column):
+    raise NotImplementedError()
+
+
+@is_bool.specialize("pandas")
+def _is_bool_pandas(column):
+    return pandas.api.types.is_bool_dtype(column)
+
+
+@is_bool.specialize("polars")
+def _is_bool_polars(column):
+    return column.dtype == pl.Boolean
+
+
+@dispatch
+def is_numeric(column):
+    raise NotImplementedError()
+
+
+@is_numeric.specialize("pandas")
+def _is_numeric_pandas(column):
+    # polars and pandas disagree about whether Booleans are numbers
```

Comment:
here which should we pick for `is_numeric(Boolean)` ? pandas says yes, polars says no.
for example in the TableVectorizer do we want to apply the `numeric_transformer` to boolean columns?

for example, in a situation where the numeric transformer is simple cast to float32 and then we pass the output of the TableVectorizer to a scikit-learn estimators I think it would make sense to do the same thing for boolean columns